treeherder/webapp/api/hash.py

```diff
@@ -19,7 +20,19 @@ def tocommit(self, request, project):
             return Response(data=query_params.errors, status=HTTP_400_BAD_REQUEST)
         newhash = query_params.validated_data["newhash"]
         basehash = query_params.validated_data["basehash"]
-        newpush = Commit.objects.filter(comments__contains=newhash).first()
-        basepush = Commit.objects.filter(comments__contains=basehash).first()
-        query_params.validate_pushes(newpush, newhash, basepush, basehash)
+        newhashdate = query_params.validated_data["newhashdate"]
+        basehashdate = query_params.validated_data["basehashdate"]
+        newpush = Commit.objects.filter(
+            comments__contains=newhash,
+            push__repository=4,
```

Comment:
It would be better to search by name, since it's not readable enough to have the id hardcoded as you did here.

Reply:
Added a constant that is used to search only try, let me know if there is anything else @beatrice-acasandrei !